app/src/androidTest/java/net/skyscanner/backpack/BpkSnapshotTest.kt

```diff
@@ -76,4 +86,37 @@ open class BpkSnapshotTest {
     testName: String = TestNameDetector.getTestName(),
   ): String =
     "${testClass.removePrefix("net.skyscanner.backpack.")}_$testName"
+
+  private fun wrapMeasuredViewWithBackground(view: View): View {
+    val result = ViewMirror(view.context, view)
+    result.setBackgroundResource(background)
+
+    ViewHelpers.setupView(result)
+      .setExactHeightDp(height)
+      .setExactWidthDp(width)
+      .layout()
+
+    return result
+  }
+
+  // adding view to FrameLayout or creating custom ViewGroup breaks many tests for some reason.
```

Comment:
interesting - what were the errors? did you try setting the layoutparams when adding the view?

Reply:
Yes, tried a lot. The errors might look like incorrect layout (aligning counter in Nudget component on the top but no centre), cutting off some content (TextField cuts the top and bottom parts) etc.

Reply:
super weird!